src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -11913,7 +11913,7 @@ private async Task WriteTextFeed(TextDataFeed feed, Encoding encoding, bool need
 
             encoding = encoding ?? TextDataFeed.DefaultEncoding;
 
-            using (ConstrainedTextWriter writer = new ConstrainedTextWriter(new StreamWriter(new TdsOutputStream(this, stateObj, null), encoding), size))
+            using (ConstrainedTextWriter writer = encoding == Encoding.UTF8 ? new ConstrainedTextWriter(new StreamWriter(new TdsOutputStream(this, stateObj, null)), size) : new ConstrainedTextWriter(new StreamWriter(new TdsOutputStream(this, stateObj, null), encoding), size))
```

Comment:
Is this change needed?
These should already be equivalent:
```
new StreamWriter(new TdsOutputStream(this, stateObj, null))
```
```
new StreamWriter(new TdsOutputStream(this, stateObj, null), Encoding.UTF8)
```

Reply:
[StreamWriter ](https://learn.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=net-8.0#system-io-streamwriter-ctor(system-io-stream)) without the encoding parameter uses UTF8 encoding with no BOM by default which is what the server also expects for json. In case the encoding is specified explicitly, we need to remove the BOM explicitly before sending the data to the server.
```csharp
        public StreamWriter(Stream stream)
            : this(stream, UTF8NoBOM, DefaultBufferSize, false)
        {
        }

        public StreamWriter(Stream stream, Encoding encoding)
            : this(stream, encoding, DefaultBufferSize, false)
        {
        }
```

Reply:
I see. So why not pass in UTF8Encoding.UTF8Encoding() https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=net-8.0#system-text-utf8encoding-ctor when we call WriteTextFeed?

Reply:
Done